src/library_exceptions.js

```diff
@@ -428,6 +428,20 @@ var LibraryExceptions = {
     catchInfo.free();
     {{{ makeThrow('ptr') }}}
   },
+
+#if !DISABLE_EXCEPTION_CATCHING
+  $formatException__deps: ["emscripten_format_exception", "$withStackSave", "free"],
+  $formatException: function(excPtr){
+      return withStackSave(function(){
+      var result_ptr = stackAlloc({{{ POINTER_SIZE }}});
+      _emscripten_format_exception(result_ptr, excPtr);
```

Comment:
Can we just have this function return the `result_ptr` and therefore avoid the stack allocation and the `withStackSave`?

The return value could be "a newly allocated string, or NULL on failure"

Reply:
Sounds good.